### PR TITLE
provider/cloudwatch_alarm: do not convert arns into autoscaling groups

### PR DESCRIPTION
**Pull request by fsouza**
The way I detect that it's a arn is by checking whether the string
starts with arn:aws. There are probably better ways of doing that. We
could be more specific and support only certain kinds of arns, for
example.

I failed to change/prepare a test for this change. I'd love to include a
test in this change, if someone could point me to the right direction.

This is related to #227.

### CODE REVIEW

**Comment by DavidS** [comment]
I think I understand what's going on here: the API and the manifests use different concepts to refer to alarms, and the original code seems to be confused about that. In the same vein I'm not convinced that the conditionals are required here. e.g. `alarm_to_hash` should just know what gets passed in (values from the API). The original code is also passing all tests (in spec/acceptance/autoscaling_spec.rb).

I'll be offline for the next two weeks, so I can't currently give this as much attention as it should have. If you could have a look and try to build a acceptance test that demonstrates the cases inwhen the original error happens? Or, even just modify the current ASG tests to hit that issue, would already be a great step forward, so I can pick it up when I return.

**Comment by fsouza** [comment]
@DavidS alarm_to_hash makes a call to describe autoscaling policies , and AWS fails saying that the provided value is not a valid asg identifier. The test suite mocks AWS API, it isn't capable of ensuring that the library behaves correctly with the actual API. I wasn't able to add a new test case with the error response form the AWS API, but if you can point me to the right direction, I can have a look.

As described in https://github.com/puppetlabs/puppetlabs-aws/issues/227#issuecomment-205213335, the error messages says "Malformed policy ARN.".

File: lib/puppet/provider/cloudwatch_alarm/v2.rb [hunk 1]
```diff
@@ -33,10 +33,15 @@ def self.prefetch(resources)
   read_only(:region, :alarm_actions)
 
   def self.alarm_to_hash(region, alarm)
-    response = autoscaling_client(region).describe_policies(
-      policy_names: alarm.alarm_actions
-    )
-    actions = response.scaling_policies.collect(&:policy_name)
+    actions = alarm.alarm_actions
+
+    unless actions.first.start_with?('arn:aws')
```
Comment (hunner) [inline lib/puppet/provider/cloudwatch_alarm/v2.rb]: Is it reasonable to only inspect the first action, or should all actions be inspected for `arn:aws` when collecting the `policy_name`?

Comment (zleswomp) [inline lib/puppet/provider/cloudwatch_alarm/v2.rb]: @fsouza I also have this question.

Comment (fsouza) [inline lib/puppet/provider/cloudwatch_alarm/v2.rb]: Hey, yes, we should probably inspect all actions. What should the code do in this case? Collect all the actions that are not arn as pass as parameter  to describe_policies? Then merge "actions" back?

I can work on that if we agree with this solution. Sorry for taking almost one month to respond :(

**Pull request closed without merging**